mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -57,22 +69,21 @@ class ChiSqSelectorModel @Since("1.3.0") (
    * Preserves the order of filtered features the same as their indices are stored.
    * Might be moved to Vector as .slice
    * @param features vector
-   * @param filterIndices indices of features to filter
+   * @param filterIndices indices of features to filter, must be ordered asc
    */
   private def compress(features: Vector, filterIndices: Array[Int]): Vector = {
-    val orderedIndices = filterIndices.sorted
```

Comment:
@srowen Yes, accept any input, sort it and store the sorted array will spend same computation with my change. But if we remove this requirement, it will introduce incompatible change which should update ```MimaExcludes``` to escape. I revert the change of #14597 did for ```MimaExcludes```. I think we should not break compatibility if there is no strong requirement. Thanks.